src/validators/literal.rs

```diff
@@ -35,7 +34,7 @@ pub struct LiteralLookup<T: Debug> {
     // Catch all for hashable types like Enum and bytes (the latter only because it is seldom used)
     expected_py_dict: Option<Py<PyDict>>,
     // Catch all for unhashable types like list
-    expected_py_list: Option<Py<PyList>>,
+    expected_py_values: Option<Vec<(Py<PyAny>, usize)>>,
```

Comment:
I'm assuming this might be more performant - are there any benchmarks that reflect that?

Reply:
No benchmarks, but otherwise yes, we don't need to make Python FFI calls here to maintain this internal state :)